### PR TITLE
feat: Upgrade stream sdk

### PR DESCRIPTION
**Pull request by sfc-gh-jmichalak**
- generate object assertions
- add helpers client
- rename methods in table helpers client for consistency
- adjust and regenerate sdk
- remove fields that are not present in Snowflake
- add LastQueryId function - it's needed to test stream's AT|BEFORE
- add missing tests
- clean up integration tests
- add some external table helper functions

## Test Plan

* [x] integration tests

* [x] unit tests

## References

https://docs.snowflake.com/en/sql-reference/sql/create-stream

## TODO
Adjust SHOW field types. Add new resources for each type of stream. Adjust stream data source.

### CODE REVIEW

File: pkg/sdk/streams_validations_gen.go [hunk 1]
```diff
@@ -29,21 +27,21 @@ func (opts *CreateOnTableStreamOptions) validate() error {
 		errs = append(errs, errOneOf("CreateOnTableStreamOptions", "IfNotExists", "OrReplace"))
 	}
 	if valueSet(opts.On) {
-		if ok := exactlyOneValueSet(opts.On.At, opts.On.Before); !ok {
+		if !exactlyOneValueSet(opts.On.At, opts.On.Before) {
 			errs = append(errs, errExactlyOneOf("CreateOnTableStreamOptions.On", "At", "Before"))
 		}
 		if valueSet(opts.On.Statement) {
```
Comment (sfc-gh-jcieslak) [inline pkg/sdk/streams_validations_gen.go:33]: There should be an error when it's not set

File: pkg/sdk/testint/streams_gen_integration_test.go [hunk 2]
```diff
@@ -18,37 +19,93 @@ func TestInt_Streams(t *testing.T) {
...
-		// TODO [SNOW-1348112]: make nicer during the stream rework
-		assert.Equal(t, table.ID().FullyQualifiedName(), sdk.NewSchemaObjectIdentifierFromFullyQualifiedName(*s.TableName).FullyQualifiedName())
-		assertStream(t, s, id, "Table", "DEFAULT")
+		assertions.AssertThatObject(t, objectassert.Stream(t, id).
+			HasName(id.Name()).
+			HasDatabaseName(id.DatabaseName()).
+			HasSchemaName(id.SchemaName()).
+			HasComment("some comment").
+			HasSourceType("Table").
+			HasMode("APPEND_ONLY").
```
Comment (sfc-gh-jcieslak) [inline pkg/sdk/testint/streams_gen_integration_test.go:57]: are there any enums for sourceType and mode? Asserts could be extended to use them

File: pkg/sdk/testint/streams_gen_integration_test.go [hunk 3]
```diff
@@ -66,69 +123,82 @@ func TestInt_Streams(t *testing.T) {
...
 		stage, cleanupStage := testClientHelper().Stage.CreateStageWithDirectory(t)
 		t.Cleanup(cleanupStage)
 
 		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
-		req := sdk.NewCreateStreamOnDirectoryTableRequest(id, stage.ID()).WithComment(sdk.String("some comment"))
+		req := sdk.NewCreateOnDirectoryTableStreamRequest(id, stage.ID()).WithComment("some comment")
 		err := client.Streams.CreateOnDirectoryTable(ctx, req)
 		require.NoError(t, err)
 		t.Cleanup(func() {
 			err := client.Streams.Drop(ctx, sdk.NewDropStreamRequest(id))
```
Comment (sfc-gh-jcieslak) [inline pkg/sdk/testint/streams_gen_integration_test.go:151]: client.DropFunc

File: pkg/sdk/testint/streams_gen_integration_test.go [hunk 4]
```diff
@@ -66,69 +123,82 @@ func TestInt_Streams(t *testing.T) {
...
 		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
-		req := sdk.NewCreateStreamOnViewRequest(id, view.ID()).WithComment(sdk.String("some comment"))
+		req := sdk.NewCreateOnViewStreamRequest(id, view.ID()).
+			WithAppendOnly(true).
+			WithShowInitialRows(true).
+			WithComment("some comment")
 		err := client.Streams.CreateOnView(ctx, req)
 		require.NoError(t, err)
 		t.Cleanup(func() {
 			err := client.Streams.Drop(ctx, sdk.NewDropStreamRequest(id))
```
Comment (sfc-gh-jcieslak) [inline pkg/sdk/testint/streams_gen_integration_test.go:181]: client.DropFunc (there's more)

File: pkg/sdk/testint/streams_gen_integration_test.go (hunk 2 above)
Comment (sfc-gh-jmichalak) [reply pkg/sdk/testint/streams_gen_integration_test.go:57]: These will be added in the next PR.

File: pkg/sdk/streams_validations_gen.go (hunk 1 above)
Comment (sfc-gh-jmichalak) [reply pkg/sdk/streams_validations_gen.go:33]: But Statement is not a pointer here, so when it's not set, it fails on `!exactlyOneValueSet...`

File: pkg/sdk/testint/streams_gen_integration_test.go (hunk 3 above)
Comment (sfc-gh-jmichalak) [reply pkg/sdk/testint/streams_gen_integration_test.go]: Done.

File: pkg/sdk/testint/streams_gen_integration_test.go (hunk 4 above)
Comment (sfc-gh-jmichalak) [reply pkg/sdk/testint/streams_gen_integration_test.go]: Done.

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Pull request merged**